File: apollo-cache/src/main/java/com/apollographql/apollo/internal/cache/normalized/WriteableStore.java
package com.apollographql.apollo.internal.cache.normalized;

import com.apollographql.apollo.cache.CacheHeaders;
import com.apollographql.apollo.cache.normalized.Record;

import org.jetbrains.annotations.NotNull;

import java.util.Collection;
import java.util.Set;

public interface WriteableStore extends ReadableStore {

  Set<String> merge(@NotNull Collection<Record> recordCollection, @NotNull CacheHeaders cacheHeaders);
  Set<String> merge(Record record, @NotNull CacheHeaders cacheHeaders);
}

File: apollo-normalized-cache-api/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/internal/WriteableStore.kt
package com.apollographql.apollo3.cache.normalized.internal

import com.apollographql.apollo3.cache.CacheHeaders
import com.apollographql.apollo3.cache.normalized.Record

interface WriteableStore : ReadableStore {
  fun merge(records: Collection<Record>, cacheHeaders: CacheHeaders): Set<String>
  fun merge(record: Record, cacheHeaders: CacheHeaders): Set<String>
}
